rclcpp/include/rclcpp/context.hpp

```diff
@@ -196,29 +204,38 @@ class Context : public std::enable_shared_from_this<Context>
    * and persist on repeated init's.
    *
    * \param[in] callback the on shutdown callback to be registered
-   * \return the callback passed, for convenience when storing a passed lambda
+   * \return the created shared pointer of callback handler
    */
   RCLCPP_PUBLIC
   virtual
-  OnShutdownCallback
-  on_shutdown(OnShutdownCallback callback);
+  OnShutdownCallbackHandle::WeakPtr
+  on_shutdown(OnShutdownCallbackHandle::OnShutdownCallbackType callback);
+
+  /// Remove a on_shutdown callback handle registered with on_shutdown.
+  /**
+   * \param[in] callback_handle the handle to be removed.
+   */
+  RCLCPP_PUBLIC
+  virtual
+  void
+  remove_on_shutdown_callback(const OnShutdownCallbackHandle::WeakPtr callback_handle);
```

Comment:
Either remove the (redundant `const` on a by-value argument):

```suggestion
  remove_on_shutdown_callback(OnShutdownCallbackHandle::WeakPtr callback_handle);
```

Or make it a const-reference:

```suggestion
  remove_on_shutdown_callback(const OnShutdownCallbackHandle::WeakPtr & callback_handle);
```

Reply:
I'd prefer the latter, myself, but either is fine.

Reply:
I want to declare const-reference. But miss `&`. I will fix it.